### PR TITLE
Different matrix for preconditioner in PETSc

### PR DESCRIPTION
**Pull request by masterleinad**
Fixing Issue #1978

### CODE REVIEW

File: source/lac/petsc_solver.cc [hunk 1]
```diff
@@ -104,6 +92,18 @@ namespace PETScWrappers
         ierr = KSPSetPC (solver_data->ksp, preconditioner.get_pc());
         AssertThrow (ierr == 0, ExcPETScError(ierr));
 
+        // setting the preconditioner overwrites the used matrices.
+        // hence, we need to set the matrices after the preconditioner.
+        // the last argument is irrelevant here,
```
Comment (davydden) [inline source/lac/petsc_solver.cc]: i suppose this part of the comment (take from the original one) belongs to the first case inside `#if-else-endif`?

Comment (masterleinad) [inline source/lac/petsc_solver.cc]: Yes, of course.

Comment (davydden) [inline source/lac/petsc_solver.cc]: would you be willing to move it there? I think it reads better that way, although it's clear that it applies to the first part of in-else-endif.

**Comment by davydden** [comment]
thanks Daniel for fixing it! It looks good to me.

File: source/lac/petsc_solver.cc (hunk 1 above)
Comment (masterleinad) [inline source/lac/petsc_solver.cc]: Sure, here you go.

**Pull request merged**